nat-lab/tests/utils/connection_tracker.py

```diff
@@ -128,21 +175,67 @@ async def execute(self) -> None:
 
         await self._process.execute(stdout_callback=self.on_stdout)
 
+    async def get_all_events(self) -> List[ConntrackerEvent]:
+        """Returns a list of the reported events that matched the ConnectionTrackerConfig"""
+        if platform.system() == "Darwin":
+            return []
+        if not self._config:
+            return []
+
+        await self._synchronize()
+
+        async with self._lock:
+            return self._events
+
+    async def wait_for_event_with_tcp_state(
+        self, state: TcpState, timeout: float | None = None
+    ):
```

Comment:
2 things that I see that could be improved:
1. Easy: get rid of timeout parameter. We don't need to handle timeout ourselves this way. If someone needs a timeout on an async method they can always use `asyncio.wait_for` to wrap it.
2. More work: I think this would be nicer if it had worked as our `OutputNotifier`. Meaning you would have a method `notify_on_tcp_state(TcpState)` and later you would wait on an event which is going to be set when this tcp state is reported. This way you don't need to create this busy loop.

Reply:
Hmm good point about the timeout.
Thanks, I'll take a look into the `OutputNotifier`.